tests/providers/amazon/aws/transfers/test_s3_to_redshift.py

```diff
@@ -59,6 +59,61 @@ def test_execute(self, mock_run, mock_session):
 
         credentials_block = build_credentials_block(mock_session.return_value)
         copy_query = op._build_copy_query(credentials_block, copy_options)
+        expected_copy_query = '''
+                                COPY schema.table
+                                FROM 's3://bucket/key'
+                                with credentials
+                                'aws_access_key_id=aws_access_key_id;aws_secret_access_key=aws_secret_access_key'
+                                ;
+                             '''
+        assert_equal_ignore_multiple_spaces(self, expected_copy_query, copy_query)
+
+        assert mock_run.call_count == 1
+        assert access_key in copy_query
+        assert secret_key in copy_query
+        assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)
```

Comment:
here we have access to the query, so we can drop of invoking private methods and compare this value. Thus, this test will still check implementations, but will depend less on the internal details.

Reply:
Thanks for the comment ;)
So would it be nicer to check execution with plain text rather than invoking private method `op._build_copy_query` but ike this? 
From invoking private method
```python
        copy_query = op._build_copy_query(credentials_block, copy_options)
        expected_copy_query = '''
                                COPY schema.table
                                FROM 's3://bucket/key'
                                with credentials
                                'aws_access_key_id=aws_access_key_id;aws_secret_access_key=aws_secret_access_key'
                                ;
                             '''
        assert_equal_ignore_multiple_spaces(self, expected_copy_query, copy_query)

        assert mock_run.call_count == 1
        assert access_key in copy_query
        assert secret_key in copy_query
        assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)
```
To check with platin text
```python
        copy_query = '''
                                COPY schema.table
                                FROM 's3://bucket/key'
                                with credentials
                                'aws_access_key_id=aws_access_key_id;aws_secret_access_key=aws_secret_access_key'
                                ;
                             '''
        assert mock_run.call_count == 1
        assert access_key in copy_query
        assert secret_key in copy_query
        assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)
```

Reply:
Yes. Exactly. We should only check the places where the class is connecting to another class, not its internal details.

Reply:
Done :) 
Changed all copy_query to plain text
Thanks for all the comment @mik-laj